server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -33,39 +33,206 @@ import org.log4s.getLogger
     */
-  final case class Config[F[_]](
-      systemPath: String,
-      pathCollector: PathCollector[F],
-      pathPrefix: String,
-      bufferSize: Int,
-      cacheStrategy: CacheStrategy[F],
-  )
+  final class Config[F[_]](
+      val systemPath: String,
+      val fs2PathCollector: Fs2PathCollector[F],
+      val pathPrefix: String,
+      val cacheStrategy: CacheStrategy[F],
+      val bufferSize: Int,
+  ) extends Product
+      with Serializable
+      with Equals {
+
+    /** For binary compatibility.
+      * @param systemPath path prefix to the folder from which content will be served
+      * @param pathPrefix prefix of Uri from which content will be served
+      * @param pathCollector function that performs the work of collecting the file or rendering the directory into a response.
+      * @param bufferSize buffer size to use for internal read buffers
+      * @param cacheStrategy strategy to use for caching purposes. Default to no caching.
+      */
+    @deprecated("use the constructor with fs2PathCollector", "0.23.8")
+    def this(
+        systemPath: String,
+        pathCollector: PathCollector[F],
+        pathPrefix: String,
+        bufferSize: Int,
+        cacheStrategy: CacheStrategy[F],
+    ) =
+      this(
+        systemPath,
+        (path: Path, config: Config[F], request: Request[F]) =>
+          pathCollector(
+            path.toNioPath.toFile,
+            config,
+            request,
+          ),
+        pathPrefix,
+        cacheStrategy,
+        bufferSize,
+      )
+
+    /** For binary compatibility.
+      * @return an instance of PathCollector[F] created (converted) from fs2PathCollector
+      */
+    @deprecated("use fs2PathCollector", "0.23.8")
+    def pathCollector: PathCollector[F] = (file, config, request) =>
+      fs2PathCollector(
+        Path.fromNioPath(file.toPath()),
+        config,
+        request,
+      )
+
+    @deprecated(
+      "Config is no longer a case class. The copy method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    def copy(
+        systemPath: String = this.systemPath,
+        @nowarn pathCollector: PathCollector[F] = this.pathCollector,
+        pathPrefix: String = this.pathPrefix,
+        bufferSize: Int = this.bufferSize,
+        cacheStrategy: CacheStrategy[F] = this.cacheStrategy,
+    ): Config[F] = new Config[F](
+      systemPath,
+      fs2PathCollector,
+      pathPrefix,
+      cacheStrategy,
+      bufferSize,
+    )
+
+    @deprecated(
+      "Config is no longer a case class. The productArity method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    override def productArity: Int = 5
+
+    @deprecated(
+      "Config is no longer a case class. The productElement method is provided for binary compatibility.",
+      "0.23.8",
+    )
+    override def productElement(n: Int): Any = n match {
+      case 0 => systemPath
+      case 1 => fs2PathCollector
```

Comment:
Since that's what the old API would return :)
```suggestion
      case 1 => pathCollector
```

Reply:
Yeah, that's what I had in my mind when typing this. My typing devices decided to do something else apparently  =)